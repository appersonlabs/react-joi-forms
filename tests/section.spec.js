var renderer = require("react-test-renderer");
var Form = require("../src/index.js").JoiForm;
var FormSection = require("../src/index.js").FormSection;
var React = require("react");

var Joi = require("joi-browser");

describe("FormSection", () => {
    test("Should use camel cased label for name when no name provided", done => {
        var joyStuff = [Joi.string().label("First Name")];
        var FormComponent = renderer.create(
            <Form
                schema={joyStuff}
                onSubmit={(err, values) => {
                    expect(values).to.exist;
                    expect(values.firstName).to.exist;

                    done();
                }}
            />
        );
        let tree = FormComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Should use name when provided", done => {
        var joyStuff = [
            Joi.string().label("First Name").meta({ name: "funky" })
        ];
        var FormComponent = renderer.create(
            <Form
                schema={joyStuff}
                onSubmit={(err, values) => {
                    expect(values).to.exist;
                    expect(values.funky).to.exist;
                    expect(values.firstName).to.not.exist;

                    done();
                }}
            />
        );

        let tree = FormComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Should create a form containing one section", () => {
        var joiSchema = [
            Joi.string().label("field one"),
            Joi.string().label("field two")
        ];
        var FormComponent = renderer.create(
            <Form>
                <FormSection schema={joiSchema} />
            </Form>
        );
        let tree = FormComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Should create a form containing two sections but no tags", () => {
        var joiSchema = [
            Joi.string().label("field one"),
            Joi.string().label("field two")
        ];
        var FormComponent = renderer.create(
            <Form schema={joiSchema}>
                <div>
                    <FormSection />
                </div>
                <div>
                    <FormSection />
                </div>
            </Form>
        );
        let tree = FormComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Should create a form containing two sections w/tags", () => {
        var joiSchema = [
            Joi.string().label("field one").tags("section1"),
            Joi.string().label("field two").tags("section2")
        ];
        var FormComponent = renderer.create(
            <Form schema={joiSchema}>
                <div>
                    <FormSection tag="section1" />
                </div>
                <div>
                    <FormSection tag="section2" />
                </div>
            </Form>
        );
        let tree = FormComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Should create a form with a section containing 0 fields", () => {
        var joiSchema = [
            Joi.string().label("field one").tags("section1"),
            Joi.string().label("field one").tags("section2")
        ];
        var FormComponent = renderer.create(
            <Form schema={joiSchema}>
                <FormSection tag="foo" />
            </Form>
        );
        let tree = FormComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Should create a form containing two sections but both values should be returned", done => {
        var joiSchema = [
            Joi.string().label("field one").tags("section1"),
            Joi.string().label("field two").tags("section2")
        ];
        var FormComponent = renderer.create(
            <Form
                schema={joiSchema}
                onSubmit={function(err, data) {
                    expect(err).to.not.exist;
                    expect(data).to.exist;
                    expect(data.fieldOne).to.equal("giraffe");
                    expect(data.fieldTwo).to.equal("snake");

                    done();
                }}>
                <div>
                    <FormSection tag="section1" />
                </div>
                <div>
                    <FormSection tag="section2" />
                </div>
            </Form>
        );
        let tree = FormComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Should create a form containing two sections but both values should be returned even with a component layout", done => {
        var Grid = React.createClass({
            render() {
                return (
                    <div>
                        {this.props.children}
                    </div>
                );
            }
        });

        var joiSchema = [
            Joi.string().label("field one").tags("section1"),
            Joi.string().label("field two").tags("section2")
        ];
        var FormComponent = renderer.create(
            <Form
                schema={joiSchema}
                onSubmit={function(err, data) {
                    expect(err).to.not.exist;
                    expect(data).to.exist;
                    expect(data.fieldOne).to.equal("giraffe");
                    expect(data.fieldTwo).to.equal("snake");

                    done();
                }}>
                <Grid>
                    <FormSection tag="section1" />
                </Grid>
                <Grid>
                    <FormSection tag="section2" />
                </Grid>
            </Form>
        );
        let tree = FormComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
