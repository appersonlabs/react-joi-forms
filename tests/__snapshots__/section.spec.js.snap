// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSection Should create a form containing one section 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    style={undefined}
  />
</form>
`;

exports[`FormSection Should create a form containing two sections but both values should be returned 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <div
      style={undefined}
    >
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
    </div>
  </div>
  <div>
    <div
      style={undefined}
    >
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
    </div>
  </div>
</form>
`;

exports[`FormSection Should create a form containing two sections but both values should be returned even with a component layout 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <div
      style={undefined}
    >
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
    </div>
  </div>
  <div>
    <div
      style={undefined}
    >
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
    </div>
  </div>
</form>
`;

exports[`FormSection Should create a form containing two sections but no tags 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <div
      style={undefined}
    >
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
    </div>
  </div>
  <div>
    <div
      style={undefined}
    >
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
    </div>
  </div>
</form>
`;

exports[`FormSection Should create a form containing two sections w/tags 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <div
      style={undefined}
    >
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
    </div>
  </div>
  <div>
    <div
      style={undefined}
    >
      <span>
        Input type 
        text
         does not have a defined component type
      </span>
    </div>
  </div>
</form>
`;

exports[`FormSection Should create a form with a section containing 0 fields 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    style={undefined}
  />
</form>
`;

exports[`FormSection Should use camel cased label for name when no name provided 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    style={undefined}
  >
    <span>
      Input type 
      text
       does not have a defined component type
    </span>
  </div>
</form>
`;

exports[`FormSection Should use name when provided 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    style={undefined}
  >
    <span>
      Input type 
      text
       does not have a defined component type
    </span>
  </div>
</form>
`;
